dll/win32/shell32/COpenWithMenu.cpp

```diff
@@ -217,6 +217,12 @@ LPCWSTR COpenWithList::GetName(SApp *pApp)
         {
             WARN("Failed to load %ls info\n", pApp->wszFilename);
             StringCbCopyW(pApp->wszName, sizeof(pApp->wszName), pApp->wszFilename);
+
+            WCHAR wszPath[MAX_PATH];
+            if (!GetPathFromCmd(wszPath, pApp->wszCmd))
+            {
+                return NULL;
```

Comment:
Actually I wonder why those missing applications are _not_ filtered already much before those functions (GetName(), and the other ones that call GetName()) are called; in other words, I wonder why `SApp` objects have been constructed for those missing applications....

Reply:
Most of them (except any added from the Browse dialog later) come from the registry.

Reply:
But is the `GetName` function supposed to filter them, or someone else?

Reply:
Same question, and it was part of the description of my original ticket

Reply:
> But is the `GetName` function supposed to filter them, or someone else?

You could in theory move the validation to the part where it loads from the registry but that would make the code less generic. The current code gives you some options to filter away hidden and/or "not recommended" items without having to hit the disk during enumeration.

I suppose the function should be called something like `LoadName` or something like that, it is not a simple getter and it only gets called once per item (by the open with dialog or by the open with menu).